File: tests/fixtures/swift/basics.swift
import Foundation

struct User: Codable {
    let id: Int
    var name: String
    var email: String?
}

class UserService {
    func fetchUser(id: Int) async throws -> User {
        let url = URL(string: "https://api.example.com/users/\(id)")!
        let (data, _) = try await URLSession.shared.data(from: url)
        return try JSONDecoder().decode(User.self, from: data)
    }
}

File: tests/fixtures/scala/basics.scala
package com.example

import scala.concurrent.{Future, ExecutionContext}
import scala.util.{Try, Success, Failure}

case class User(id: Long, name: String, email: Option[String])

object UserService {
  def findById(id: Long)(implicit ec: ExecutionContext): Future[Option[User]] = Future {
    Some(User(id, "Alice", Some("alice@example.com")))
  }
}